adapters/bluesea/bluesea.go

```diff
@@ -0,0 +1,158 @@
+	impCount := len(request.Imp)
+
+	if impCount == 0 {
+		err := &errortypes.BadInput{
+			Message: "Empty Imp objects",
+		}
+		return nil, []error{err}
+	}
+
+	requestDatas := make([]*adapters.RequestData, 0, impCount)
+	errs := make([]error, 0, impCount)
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	for _, imp := range request.Imp {
+		blueseaImpExt, err := extraImpExt(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		reqJson, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		queryParams := url.Values{}
+		queryParams.Add("pubid", blueseaImpExt.PubId)
+		queryParams.Add("token", blueseaImpExt.Token)
+		queryString := queryParams.Encode()
+		requestData := &adapters.RequestData{
+			Method:  "POST",
+			Uri:     fmt.Sprintf("%s?%s", a.endpoint, queryString),
+			Body:    reqJson,
+			Headers: headers,
+		}
+		requestDatas = append(requestDatas, requestData)
+	}
+	return requestDatas, errs
+}
+
+func extraImpExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpBluesea, error) {
+	var impExt adapters.ExtImpBidder
+	var blueseaImpExt openrtb_ext.ExtImpBluesea
+
+	err := json.Unmarshal(imp.Ext, &impExt)
+	if err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("Error in parsing imp.ext. err = %v, imp.ext = %v", err.Error(), string(imp.Ext)),
+		}
+	}
+
+	err = json.Unmarshal(impExt.Bidder, &blueseaImpExt)
+	if err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("Error in parsing imp.ext.bidder. err = %v, bidder = %v", err.Error(), string(impExt.Bidder)),
+		}
+	}
+	if len(blueseaImpExt.PubId) == 0 || len(blueseaImpExt.Token) == 0 {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("Error in parsing imp.ext.bidder, empty pubid or token"),
+		}
+	}
+	return &blueseaImpExt, nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var blueseaResponse openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &blueseaResponse); err != nil {
+		return nil, []error{fmt.Errorf("Error in parsing bidresponse body")}
+	}
+
+	var errs []error
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	if blueseaResponse.Cur != "" {
+		bidResponse.Currency = blueseaResponse.Cur
+	}
+	for _, seatBid := range blueseaResponse.SeatBid {
+		for _, bid := range seatBid.Bid {
+
+			bidType, err := getMediaTypeForBid(&bid)
+
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			b := &adapters.TypedBid{
+				Bid:     &bid,
```

Comment:
Incorrect assignment made to `Bid`  Recommended to either create copy of `bid`
```
          for _, seatBid := range response.SeatBid {
            for _, bid := range seatBids.Bid {
              ...
              seatBid := bid
              responseBid := &adapters.TypedBid{
                Bid: &seatBid,
                ...
              }
              ...
            }
          }
```


Alternatively make use of `seatBids.Bid` loop index
```
          for _, seatBid := range response.SeatBid {
            for i, bid := range seatBids.Bid {
              ...
              responseBid := &adapters.TypedBid{
                Bid: &seatBids.Bid[i],
                ...
              }
              ...
            }
          }
```